src/main/java/io/openliberty/tools/intellij/actions/LibertyGeneralAction.java

```diff
@@ -159,14 +159,14 @@ protected void notifyError(String errMsg) {
     }
 
     /**
-     * Returns the IntelliJ terminal widget for the corresponding Liberty module
+     * Returns the Terminal widget for the corresponding Liberty module
      *
-     * @param createWidget create terminal widget if it does not already exist
+     * @param createWidget create Terminal widget if it does not already exist
      * @return ShellTerminalWidget
      */
     protected ShellTerminalWidget getTerminalWidget(boolean createWidget) {
-        ShellTerminalWidget widget = LibertyProjectUtil.getTerminalWidget(project, projectName, createWidget);
-        if (widget == null) {
+        ShellTerminalWidget widget = LibertyProjectUtil.getTerminalWidget(project, libertyModule, createWidget);
+        if (widget == null || (!createWidget && !widget.hasRunningCommands())) {
```

Comment:
suggestion to add a brief comment explaining the "createWidget" and "hasRunningCommands" logic

Reply:
Addressed